scripts/tools/zap/zap_execution.py

```diff
@@ -47,11 +56,15 @@ def __init__(self):
             os.environ['ZAP_SKIP_REAL_VERSION'] = '1'
             self.check_version = False  # cannot check versions without dirtying tree
         elif 'ZAP_INSTALL_PATH' in os.environ:
-            self.zap_start = [os.path.join(
-                os.environ['ZAP_INSTALL_PATH'], 'zap-cli')]
+            self.zap_start = [os.path.join(os.environ['ZAP_INSTALL_PATH'], 'zap-cli')]
             self.working_directory = None
         else:
-            self.zap_start = ['zap-cli']
+            # If we're running within a Python venv, check for a zap-cli binary there
```

Comment:
why would a python venv have zap-cli in its path? zap is not python, so this seems unexpected or I do not understand the comment.

If we change zap finding logic, we probably have to update our documentation as well about the logic and reasoning. Currently we say "if the other things are not set, then zap-cli must be in $PATH". This checking logic overrides $PATH.

Reply:
Zap isn’t Python, but the Python venv has a bin directory that’s convenient to place other tools in as well (at least optionally), treating it as a generic build env. This is where the configure script downloads zap-cli to. I can update the docs / banner we print to document this (not sure this would be somewhere where someone would manually install zap though).

I think the key thing is that both the pigweed bootstrap and the configure script place zap-cli so that code gen just works.

The instructions we currently print are geared towards a manual installation. Maybe we first bullet should point towards bootstrap and configure since most users shouldn’t need to install by hand.

In terms of implementation, prepending the venv bin to PATH might capture the intent better?

Reply:
Hm maybe I can do the prepending in a `sitecustomize` file in the venv and leave the zap script as is, that might be the tidiest solution.

Reply:
I've reverted the change to zap_execution.py and instead made the venv fully auto-activate (including prepending its bin directory to PATH). This makes running a script via the venv python behave the same as if the activate script had been sourced first.